api/migrations/20230207224139_mongoose-adding-funding-category.js

```diff
@@ -133,9 +134,9 @@ export const up = async () => {
           ...apd,
           apdType
         });
-        return getBudgetModel(apdType).replaceOne(
+        return Budget(apdType).replaceOne(
           { _id: apd.budget },
-          { ...budget }
+          { ...budget, __t: BUDGET_TYPE[apdType] }
```

Comment:
is this why it is/was failing?

Reply:
I think so because I think it wasn't properly setting the discrimination so it was only saving what is in the base budget